Adding some more context to docs (#15)
minio/hperf: README.md

Hperf was made to test networks in large infrastructure. It's highly scalable and cabaple of running parallel tests over
a long period of time. 

## Common use cases
- Debugging link/nic MTU issues
- Optimizing throughput speed for specific payload/buffer sizes
- Finding servers that present latency on the application level when ping is showing no latency
- Testing overall network throughput
- Testing server to server connectivity 

## Core concepts
### The hperf binary
The binary can act as both client and server.

### Client
The client part of hperf is responsible for orchestrating the servers. Its only job is to send commands to the
servers and receive incremental stats updates. It can be executed from any machine that can talk to the servers.

### Servers
Servers are the machines we are testing. To launch the hperf command in servers mode, simply use the `server` command:
NOTE: `server` is the only command you can execute on the servers. All other commands are executed from the client.
```bash
$ ./hperf server --help
```

> This command will start an API and websocket on the given `--address` and save test results to `--storage-path`. 

Reviewer: ```suggestion
This command will start an API and websocket on the given `--address` and save test results to `--storage-path`. 

Do not expose this port to public traffic.
```

### The listen command

> Hperf has been modified to work very differently than how it was before. The new version can run tests without a

Reviewer: Maybe just focus on how it is now... I don't know how it was before (and frankly don't care) Maybe add a separate section on it, if needed.

Also nothing here really explains what `listen` actually does, AFAICT.

Reviewer 2: No we should just document what is latest, no one knows what hperf was before.

specific `client` needing to be constantly connected. Once the `client` has started a test, the `client` can 
easily exit without interrupting the test stopping.

Any `client` can hook into the list test at runtime using the `--id` of the test. There can even be multiple `clients`
listening to the same test.

Example:
```bash

> $ ./hperf listen --hosts 10.10.1.{2...10} --id [TEST_ID]

Reviewer: Maybe post more, so I can see how ID is printed and how to reconnect after ctrl-c.

2. Run hperf help to see available commands, flags and examples
   - The `--hosts` and `--id` flags are especially important to understand
```bash
$ ./hperf --help
$ ./hperf [command] --help
```

> ```

Reviewer: ```suggestion
```

--hosts: Hosts is where we determine which machines we will send the current command to. The hosts parameter supports
the same ellipsis pattern as minio and also a comma separate list of hosts as well as a file: input. The file expects a
host per file line.

Examples:
./hperf [command] --hosts 1.1.1.1,2.2.2.2
./hperf [command] --hosts 1.1.1.{1...100}

> ./hperf [command] --hosts file:/home/user/hosts

Reviewer: This should have triple quotes only. Can't suggest it.

Additionally, if the `server` command was executed with a custom address + port, the port can be specified using `--port`.

> --id: is the ID used when starting tests, listening to tests, or fetching test results. Be careful not to re-use the

Reviewer: ```suggestion
`--id` is the ID used when starting tests, listening to tests, or fetching test results. Be careful not to re-use the
```

ID's if you care about fetching results at a later date.

> ```

Reviewer: ```suggestion
```

 - CPU (CPUUsed): 
   - Total memory in use (total for all time)

## Example: 20 second HTTP payload transfer test using multiple sockets

> This test will use 12 concurrent workers(concurrency) to send http requests with a payload without any timeouts inbetween requests.

Reviewer: ```suggestion
This test will use 12 concurrent workers to send http requests with a payload without any timeout between requests.
```

Much like a bandwidth test, but it will also test server behaviour when multiple sockets are being created and closed.
```
$ ./hperf requests --hosts file:./hosts --id http-test-1 --duration 20 --concurrency 12
```

## Example: 20 second HTTP payload transfer test using a stream

> This will perform a 20 second bandwidth test with 12 concurrent HTTP streams.

Reviewer: ```suggestion
This will perform a 20 second bandwidth test with 12 concurrent HTTP streams (concurrently).
```
Nah.. Just messing with you :D

```
$ ./hperf bandwidth --hosts file:./hosts --id http-test-2 --duration 20 --concurrency 12
```

## Example: 5 Minute latency test using a 2000 Byte buffer, with a delay of 50ms between requests

> This test will send a single round trip request between servers to test base latency and reachability 

Reviewer: ```suggestion
This test will send a single round trip request between servers to test base latency and reachability:
```
